src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FallThroughCheck.java

```diff
@@ -486,25 +495,25 @@ private boolean checkSynchronized(final DetailAST synchronizedAst, boolean useBr
      * </pre>
      *
      * @param currentCase AST of the case that falls through to the next case.
-     * @param nextCase AST of the next case.
      * @return True if a relief comment was found
      */
-    private boolean hasFallThroughComment(DetailAST currentCase, DetailAST nextCase) {
-        boolean allThroughComment = false;
-        final int endLineNo = nextCase.getLineNo();
-
-        if (matchesComment(reliefPattern, endLineNo)) {
-            allThroughComment = true;
+    private boolean hasFallThroughComment(DetailAST currentCase) {
+        final DetailAST nextCase = currentCase.getNextSibling();
+        DetailAST commentAst;
+        if (nextCase.getType() == TokenTypes.CASE_GROUP) {
+            commentAst = nextCase.getFirstChild();
         }
         else {
-            final int startLineNo = currentCase.getLineNo();
-            for (int i = endLineNo - 2; i > startLineNo - 1; i--) {
-                final int[] line = getLineCodePoints(i);
-                if (!CodePointUtil.isBlank(line)) {
-                    allThroughComment = matchesComment(reliefPattern, i + 1);
-                    break;
-                }
+            commentAst = currentCase;
+        }
+        boolean allThroughComment = false;
+        while (commentAst != null) {
+            if (commentAst.getFirstChild() != null
+                    && matchesComment(reliefPattern, commentAst.getFirstChild().getText())) {
+                allThroughComment = true;
+                break;
             }
+            commentAst = commentAst.getNextSibling();
```

Comment:
case might have a lot of tokens, you need operate on comments, so you need `next sibling of types X or Y`

Reply:
I have made mistake in naming commentAst must ast i don't know why i have just written commentAST

Reply:
```
class Test {

        void methodFallThru(int i, int j, boolean cond) {
        while (true) {
            switch (i) {
                case -1: // FALLTHRU
                case 0: // ok
                case 34: 
                  // fallthru
                i++;
            default: break;
         }
      }
   }
}

<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="FallThrough">
        </module>
    </module>
</module>

kevin@inspiron-15-5510:~/Desktop/check_style$ java -jar /home/kevin/Downloads/checkstyle-10.9.3-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /home/kevin/Desktop/check_style/Test.java:11:13: Fall through from previous branch of the switch statement. [FallThrough]
Audit done.
Checkstyle ends with 1 errors.
kevin@inspiron-15-5510:~/Desktop/check_style$ 

```

this is showing violation which is not expected as from latest release this type of test case is not available so my changes are also not supporting this

Reply:
Name all variables meaningfully and add support for this case

Reply:
done